main.cc

```diff
@@ -1793,8 +1793,10 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
             // Set up group0 service earlier since it is needed by group0 setup just below
             ss.local().set_group0(group0_service, raft_topology_change_enabled);
```

Comment:
> Also we remove the load_initial_raft_address_map function from raft_group0 as it seems redundant. When start_server_for_group0 is called from setup_group0_if_exist gossiper is not initialized yet, so this call has no effect.

But `start_server_for_group0` is also called from the other `setup_group0` -> `join_group0`, that's where it call is needed i.e. when we join the cluster. Not when we restart.

"seems redundant" is not a good enough reason to remove something, there was a reason it was added. The reason is that we might miss all gossiper notifications about endpoints that are delivered by gossiper before we subscribe to it.

Reply:
> I should have given a more detailed reasoning in the commit message, updated it.

The commit message looks the same to me as it was.

I would explain why it's also no longer necessary on the other path where we start server for group -- we subscribe to gossiper before starting gossiping, so we won't miss any gossiper notifications.

Reply:
> The commit message looks the same to me as it was.

Idk what's happening, sometimes git just refuses to do what I want. I clearly remember how I updated the commit message.

We do this again.